repo.go

```diff
@@ -695,6 +696,32 @@ func (r *Repo) AddTargets(paths []string, custom json.RawMessage) error {
 	return r.AddTargetsWithExpires(paths, custom, data.DefaultExpires("targets"))
 }
 
+func (r *Repo) AddDigestTargets(digest string, digestAlg string, length int64, custom json.RawMessage, path string) error {
+	expires := data.DefaultExpires("targets")
+
+	// TODO: support delegated targets
+	t, err := r.targets()
+	if err != nil {
+		return err
+	}
+
+	meta := data.FileMeta{Length: length, Hashes: make(data.Hashes, 1)}
+	meta.Hashes[digestAlg], err = hex.DecodeString(digest)
+	if err != nil {
+		return err
+	}
+
+	if len(custom) > 0 {
+		meta.Custom = &custom
+	} else if t, ok := t.Targets[path]; ok {
+		meta.Custom = t.Custom
+	}
```

Comment:
This merging logic makes it impossible to clear the `FileMeta.Custom` field by setting it to `nil`. However it does allow setting `Custom` to `{}`. What do you think about making the `custom` argument a `*json.RawMessage` and just setting `meta.Custom = custom` rather than merging, to make all operations possible?

Reply:
I just noticed that this is a shortcoming of all the existing AddTargets functions, so no need to address here.